papermill/hdfs.py

```diff
@@ -0,0 +1,162 @@
+# -*- coding: utf-8 -*-
+"""Utilities for working with HDFS."""
+import logging
+
+from pywebhdfs.errors import FileNotFound
+from pywebhdfs.webhdfs import PyWebHdfsClient
```

Comment:
Before you get too far fixing things, I do want to point out that we should probably use pyarrow for `hdfs://` paths and pywebhdfs for `webhdfs://` paths. Not all hadoop clusters will have webhdfs installed/enabled so we'll want the register to differentiate the two.

Reply:
okay. Will make changes for pyarrow :)

Reply:
If you want to leave this branch for `webhdfs` that's fine too, just wanted to let you know there was a difference between the two.

Reply:
sure

Reply:
Added the code for supporting hdfs. PR #224 addresses this.